ops/terraform/services/common/scripts/destroy-autoscaled-nodes.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+# Terraform helper script called as a local-exec provisioner at destroy-time of the Aurora cluster
+# resource. This script will enumerate the autoscaled reader nodes within the given Aurora cluster
+# (provided by environment variable $DB_CLUSTER_ID), mark them all for deletion, and then wait
+# indefinitely for all nodes to be successfully deleted.
+
+set -euo pipefail
+
+# Be _extra_ careful about the cluster ID given by the $DB_CLUSTER_ID environment variable.
+# Specifically, remove all whitespace (which is invalid for a cluster ID) and ensure that the
+# resulting, trimmed ID is non-empty before even making any deletion attempts as the AWS CLI will
+# treat an empty cluster ID as equivalent to "give me all clusters" when calling
+# describe-db-clusters
+trimmed_cluster_id="$(echo "$DB_CLUSTER_ID" | tr -d '[:space:]')"
+if [[ -z $trimmed_cluster_id ]]; then
+  echo "DB_CLUSTER_ID must not be an empty string or whitespace"
```

Comment:
As discussed, we should try to add some check to make sure this is an ephemeral environment

Reply:
Agreed. I'll add some explicit checks on the environment here to ensure that we can only run this script within an ephemeral environment.

Reply:
Done in [b2c98bc](https://github.com/CMSgov/beneficiary-fhir-data/pull/2393/commits/b2c98bc5a9dd23feb8a62270ebfd604af436f4e8) and [8d870a5](https://github.com/CMSgov/beneficiary-fhir-data/pull/2393/commits/8d870a5aecccc4af1f55b64ef826a5014863421d).